src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -91,13 +96,22 @@ public Person toModelType() throws IllegalValueException {
         }
         final Phone modelPhone = new Phone(phone);
 
+        if (telegram == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    Telegram.class.getSimpleName()));
+        }
+        if (!Telegram.isValidTelegram(telegram)) {
+            throw new IllegalValueException(Telegram.MESSAGE_CONSTRAINTS);
+        }
+        final Telegram modelTelegram = telegram;
+
         if (email == null) {
             throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Email.class.getSimpleName()));
         }
         if (!Email.isValidEmail(email)) {
```

Comment:
I looked through the code again and it seems like this is the only place that isValidEmail() has been called. Can we just let email = source.getEmail().value and then use the original isValidEmail() to check? 
Like that we wont need 2 Validation functions, which would be confusing to differentiate. Plus, if we follow this method we need to think of 2 different names for validation functions for all our optional fields, which would not be ideal.

Reply:
in this case, do you prefer letting this code check whether it is "" or not, and if it is, it needs to set Email as Email.EMPTY_EMAIL? Because I don't think changing isValidEmail() is logical since "" is not supposed to be a valid email address. So the alternative would be to let this part of the code check if it is "" before putting it into isValidEmail(String email). My motivation behind this current approach is only to not putting "" in isValidEmail.

Reply:
Hmm, sounds good. I think this would be better than having 2 separate functions for Validation.

Reply:
alright, so I'll keep the isValidEmail() to follow the regex, and I'll do the checking here. I'll also make the storage store the String value instead of the Email object. Will do that asap.